### PR TITLE
Hotfix 2.9.1

### PR DESCRIPTION
**Pull request by pazaan**
Update `buildWizardRecords` function based on CLP code, specifically to fix insulin sensitivity and recommended values.

### CODE REVIEW

File: lib/drivers/medtronic/processData.js [hunk 1]
```diff
@@ -272,24 +272,39 @@ var buildWizardRecords = function (records) {
...
+    var carbRaw = [struct.extractByte(record.body,1) >> 2 & 1];
     carbRaw = carbRaw.concat(struct.extractByte(record.body,0));
     carbInput = struct.extractBEShort(carbRaw,0);
 
-    var carbRatioRaw = [struct.extractByte(record.body,2) & BITMASKS.RIGHT_THREE];
+    var carbRatioRaw = [0,struct.extractByte(record.body,9) >> 1 & BITMASKS.RIGHT_TWO];
+    carbRatioRaw = carbRatioRaw.concat(struct.extractByte(record.body,2));
     carbRatioRaw = carbRatioRaw.concat(struct.extractByte(record.body,3));
-    carbRatio = struct.extractBEShort(carbRatioRaw,0) / 10.0;
+    carbRatio = struct.extractBEInt(carbRatioRaw,0) / 10.0;
```
Comment (gniezen) [inline lib/drivers/medtronic/processData.js]: We should extract the carb units and divide by 1000.0 instead if it's set to exchanges.

File: lib/drivers/medtronic/processData.js [hunk 2]
```diff
@@ -272,24 +272,39 @@ var buildWizardRecords = function (records) {
...
+    var rawRecommendedCarb = [0,struct.extractByte(record.body,9) & 1];
+    rawRecommendedCarb = rawRecommendedCarb.concat(struct.extractByte(record.body,7));
+    rawRecommendedCarb = rawRecommendedCarb.concat(struct.extractByte(record.body,8));
+    recommended.carb = struct.extractBEInt(rawRecommendedCarb,0) / settings.strokesPerUnit;
 
-    units = (struct.extractByte(record.body,1) & BITMASKS.LEFT_TWO) === 0x80 ? 'mmol/L' : 'mg/dL';
+    var rawRecommendedCorrection = [struct.extractByte(record.body,9) >> 3 & 31];
+    rawRecommendedCorrection = rawRecommendedCorrection.concat(struct.extractByte(record.body,6));
+    recommended.correction = twosComplement(
+      struct.extractBEShort(rawRecommendedCorrection,0),13
```
Comment (gniezen) [inline lib/drivers/medtronic/processData.js]: Hmm, nice catch! Looks like I previously only used 11 bits here.

File: lib/drivers/medtronic/processData.js [hunk 3]
```diff
@@ -272,24 +272,39 @@ var buildWizardRecords = function (records) {
...
-    recommended.correction = twosComplement((((struct.extractByte(record.body,9) & BITMASKS.LEFT_00111) << 5) + struct.extractByte(record.body,6)),11) / settings.strokesPerUnit;
+    var isfRaw = [struct.extractByte(record.date,2) >> 6 & BITMASKS.RIGHT_TWO];
+    isfRaw = isfRaw.concat(struct.extractByte(record.body,4));
+    isf = struct.extractBEShort(isfRaw,0);
 
-    recommended.net = struct.extractBEShort(record.body,12) / settings.strokesPerUnit;
+    var rawRecommendedCarb = [0,struct.extractByte(record.body,9) & 1];
+    rawRecommendedCarb = rawRecommendedCarb.concat(struct.extractByte(record.body,7));
+    rawRecommendedCarb = rawRecommendedCarb.concat(struct.extractByte(record.body,8));
+    recommended.carb = struct.extractBEInt(rawRecommendedCarb,0) / settings.strokesPerUnit;
```
Comment (gniezen) [inline lib/drivers/medtronic/processData.js]: Huh, so this is actually a 4-byte integer, where I previously had it as a 2-byte short.

File: lib/drivers/medtronic/processData.js [hunk 4]
```diff
@@ -304,9 +304,14 @@ var buildWizardRecords = function (records) {
     rawRecommendedNet = rawRecommendedNet.concat(struct.extractByte(record.body,13));
     recommended.net = struct.extractBEInt(rawRecommendedNet,0) / settings.strokesPerUnit;
 
-    units = (struct.extractByte(record.body,1) >> 6 & BITMASKS.RIGHT_TWO) !== 2 ? 'mg/dL' : 'mmol/L';
+    bgUnits = (struct.extractByte(record.body,1) >> 6 & BITMASKS.RIGHT_TWO) !== 2 ? 'mg/dL' : 'mmol/L';
+    carbUnits = (struct.extractByte(record.body,1) >> 4 & BITMASKS.RIGHT_TWO) != 2 ? 'grams' : 'exchanges';
 
-    if(units == 'mmol/L') {
+    if(carbUnits === 'exchanges') {
```
Comment (pazaan) [inline lib/drivers/medtronic/processData.js]: Nice addition :)

**Pull request merged**